src/nvim/api/private/helpers.c

```diff
@@ -808,6 +811,17 @@ static void init_type_metadata(Dictionary *metadata)
 
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
+static void init_api_level_metadata(Dictionary *metadata)
+{
+  Dictionary version = ARRAY_DICT_INIT;
+  PUT(version, "current", INTEGER_OBJ(NVIM_API_CURRENT));
+  PUT(version, "compatibility", INTEGER_OBJ(NVIM_API_COMPATIBILITY));
+#ifdef NVIM_VERSION_PRERELEASE
+  PUT(version, "prerelease", BOOLEAN_OBJ(true));
```

Comment:
Sorry to rehash this, but I didn't really get the use of the `prerelease` field.

If we do a change on `master`, we update NVIM_API_CURRENT. Plugin authors that want to support this new functionality can just check this and know that this is a prerelease (as they know what API level was in the last Nvim release).

The newly added API parts might still change, yes, but that's the same whether you have a "prerelease" flag or know it from the incremented API level: you have to update when `master` has a breaking change.

If a new Neovim version is released with this API level (i.e. the newly added parts are now stable), you need to verify anyway that the stable API is the same as the prerelease API when you first used it.

Having a separate "prerelease" field would only make sense if you want to support both unstable API on "master before release" and the stable API "at release", no?